### ISSUE TITLE
Expiring intermediate or root certificates do not trigger non-OK (exit) state

### ISSUE THREAD

**Issue by atc0005** [atc0005/check-cert#933]
We're using the file-based bundle monitoring support to keep an eye on some intermediate certificate bundles used for authenticated content checks on select web servers.

The output correctly identifies an intermediate certificate in the bundle as "WARNING", but the exit state for the plugin continues to report OK.

We're ignoring hostname validation (this is a bundle) and SANs list validation (this is an intermediates bundle and does not contain a leaf certificate).

Needs further review.

**Comment by atc0005** [comment]
Of note:

```console
$ go run ./cmd/check_cert/ --filename /path/to/interm_bundle.pem --age-warning 60 --age-critical 30 --ignore-validation-result hostname`
WARNING: Expiration validation failed: intermediate cert "XYZ" expires next with 30d 1h remaining (until 2024-10-05 23:59:59 +0000 UTC) [checks: 2 IGNORED (Hostname, SANs List), 0 FAILED, 1 SUCCESSFUL (Expiration)]
```

Properly flagged as non-OK (`WARNING: Expiration validation failed`), but later the `Expiration` check is considered successful (`[checks: 2 IGNORED (Hostname, SANs List), 0 FAILED, 1 SUCCESSFUL (Expiration)]`).

**Comment by atc0005** [comment]
Potentially related:

- https://github.com/atc0005/check-cert/issues/397
- https://github.com/atc0005/check-cert/pull/526

**Comment by atc0005** [comment]
Changes to test:

```diff
$ git diff
diff --git a/internal/certs/validation-expiration.go b/internal/certs/validation-expiration.go
index 1e554e0..8d96a70 100644
--- a/internal/certs/validation-expiration.go
+++ b/internal/certs/validation-expiration.go
@@ -212,6 +212,17 @@ func ValidateExpiration(
                        priorityModifier = priorityModifierMinimum
                }

+       // Cover scenario of intermediates expiring (but not yet expired)
+       case hasExpiringCerts:
+               err = fmt.Errorf(
+                       "expiration validation failed: %w",
+                       ErrExpiringCertsFound,
+               )
+
+               if !validationOptions.IgnoreValidationResultExpiration {
+                       priorityModifier = priorityModifierMinimum
+               }
+
        case hasExpiredIntermediateCerts &&
                !validationOptions.IgnoreExpiredIntermediateCertificates: 
```

This occurs after checking expiration of leaf certs (expired, expiring) and before expired intermediates / root.

**Comment by atc0005** [comment]
Digging further, it does not appear that expiring intermediate or root certs are triggering a non-OK state, whether in a certificate bundle (file) or live connection.

**Comment by atc0005** [comment]
Not shown above are some prototype changes that I'm testing for the next release:

- expanded version of what is shown in https://github.com/atc0005/check-cert/issues/933#issuecomment-2332731089
- additional flags to disable validation of *expiring* intermediates and roots
  - two additional flags
